internal/monitors/docker/docker.go

```diff
@@ -225,9 +227,66 @@ func parseContainerEnvSlice(env []string) map[string]string {
 	return out
 }
 
+// EnableExtraGroups enables extra metrics that were individually turned on
+// by ExtraMetrics/ExtraGroups configuration
+func EnableExtraGroups(groupEnableMap map[string]bool, extraMetrics, extraGroups []string) EnhancedMetricsConfig {
+	groupNames := []string{groupBlkio, groupCPU, groupMemory, groupNetwork}
+
+	for _, group := range extraGroups {
+		for _, gn := range groupNames {
+			if group == gn {
+				groupEnableMap[gn] = true
```

Comment:
Modifying the input param is pretty tricky and not really recommended unless you are going for some super optimization here.

Reply:
Also, why can't this just be:
```
for _, group := range extraGroups {
    groupEnableMap[group] = true
}
```

If it is a non-existent group, that should be handled somewhere else (e.g. some kind of validation).

Reply:
was trying not to rely on user provided input for the keys, but it looks better to keep validation somewhere like you suggested.

Reply:
Please correct me if I'm wrong, but here I'm not modifying input params but just taking those as a default set of values. See https://github.com/signalfx/signalfx-agent/blob/aa3d3acbe4c7b207477ce01568f2fd86e7914cc5/internal/monitors/docker/docker.go#L90

Reply:
Well, you are modifying `groupEnableMap` which is pass in as an arg.  Since this is specific to this monitor, I would just pass in conf instead and then derive the "default set of values" from conf in this function so it is more self-contained.

Reply:
Got it. Thought you meant `input param` of the monitor not the function.